velox/expression/Expr.cpp

```diff
@@ -565,7 +565,8 @@ void Expr::eval(
   // peeling and wrapping in the sub-nodes.
   //
   // TODO: Re-work the logic of deciding when to load which field.
-  if (!hasConditionals_ || distinctFields_.size() == 1) {
+  if (!hasConditionals_ || distinctFields_.size() == 1 ||
+      shouldEvaluateSharedSubexp()) {
```

Comment:
nit: can you update the comment above to include why it is important to load if using sharedSubExprs.

It would also be valuable to add a comment above evaluateSharedSubexpr() that explains the importance of having the same set of rows whenever it is employed